htlcswitch/interceptable_switch.go

```diff
@@ -59,8 +59,7 @@ func (s *InterceptableSwitch) SetInterceptor(
 // switch, any failed packets will be returned to the provided
 // ChannelLink. The link's quit signal should be provided to allow
 // cancellation of forwarding during link shutdown.
-func (s *InterceptableSwitch) ForwardPackets(linkQuit chan struct{},
-	packets ...*htlcPacket) error {
+func (s *InterceptableSwitch) ForwardPackets(packets ...*htlcPacket) error {
```

Comment:
More background: https://github.com/lightningnetwork/lnd/commit/f84cd14b12d78c29a4395088768d8cd6d4cf875d

I realized now that `linkQuit` isn't such a problem. If we are in the resume action originating from the interceptor, we don't need to worry about deadlocks because it's a different goroutine. The `ForwardPackets` call can take a nil `linkQuit` in that case.